### PR TITLE
[OBSDEF-9417] Changes to create servicelogin shared secret and mount it in IAM and DCM

### PR DESCRIPTION
**Pull request by SeemaTahaliyani**
## Purpose
[OBSDEF-9417](https://jira.cec.lab.emc.com:8443/browse/OBSDEF-9417)

- Create shared secret at the objectscale level to be used by objectscale level services. This secret is mounted in IAM and DCM.
- Create shared secret per objectstore  to be used by the object store level service. This is mounted in object store pods as in https://eos2git.cec.lab.emc.com/ECS/ecs-flex-operator/pull/451

## PR checklist
- [x] make test passed
- [x] make build passed
- [x] helm install <chart> passed
- [ ] vSphere7 make package deployments passed
- [ ] helm upgrade <chart> passed
- [ ] helm test <release_name> passed
- [x] Passed - https://asd-ecs-jenkins.isus.emc.com/jenkins/view/ECS-Flex/job/charts-custom-ci/225/

## Testing
_Paste URL of charts-custom-ci job here_

_Paste the output of your helm install/vSphere7 deployment testing here_

### CODE REVIEW

File: ecs-cluster/templates/shared_secret.yaml [hunk 1]
```diff
@@ -0,0 +1,17 @@
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: {{ .Release.Name }}-mgmt-serviclogin-shared-secret
```
Comment (lipak2345) [inline ecs-cluster/templates/shared_secret.yaml:5]: nitpick: There is a typo in **service** (missing e)

File: ecs-cluster/templates/shared_secret.yaml [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+  labels:
+    app.kubernetes.io/name: {{.Release.Name}}
+    app.kubernetes.io/version: {{.Values.tag}}
+    app.kubernetes.io/instance: {{.Release.Name}}
+    app.kubernetes.io/managed-by: {{.Release.Service}}
+    helm.sh/chart: {{.Chart.Name}}-{{.Chart.Version | replace "+" "_"}}
+    product: objectscale
+    release: {{ .Release.Name }}
+type: Opaque
+stringData:
```
Comment (eddiepavkovic) [inline ecs-cluster/templates/shared_secret.yaml:16]: Is  this a secret that shouldn't change on upgrade?   Is so we need to add
Annotations that'll prevent it from changing

**Review by eddiepavkovic — CHANGES_REQUESTED** [review]

File: ecs-cluster/templates/shared_secret.yaml (hunk 2 above)
Comment (SeemaTahaliyani) [reply ecs-cluster/templates/shared_secret.yaml:16]: I think we will make it not change on upgrade. I need to add below annotations correct?
```
"helm.sh/hook": "pre-install"
"helm.sh/hook-delete-policy": "before-hook-creation"
```

Comment (SeemaTahaliyani) [reply ecs-cluster/templates/shared_secret.yaml:19]: @eddiepavkovic added annotations. Please  check. Thanks

File: ecs-cluster/templates/shared_secret.yaml (hunk 1 above)
Comment (SeemaTahaliyani) [reply ecs-cluster/templates/shared_secret.yaml]: Thanks :) fixed it.

**Comment by SeemaTahaliyani** [comment]
Made sure secret is created and mounted as needed by the federation svc service login

```
10:~/EMCECS # k get secret | grep shared
ecs-cluster-mgmt-servicelogin-shared-secret                       Opaque                                1      10s
mgmt-servicelogin-shared-secret                                   Opaque                                1      40m


10:~/EMCECS/charts # k exec -it objectscale-iam-c85cbfc4d-j2kcn bash
bash-4.4# cd /etc/mgmt/sharedsecret/
bash-4.4# cat key 
Bq0Qy9giIopNApzNaQvxzVIhW4Nh0rdI5PvkZksXzMJZnXknk18VAvB28u1QVEAHNJrervGKqY6Fq5OJMXF88VOT0YgPhgWL18EHzaEMy9mIwPRVatxzfHouSBYHFZ5xbash-4.4# ^C
bash-4.4# exit
exit
command terminated with exit code 130
10:~/EMCECS/charts # k exec -it objectscale-dcm-64d8f4499d-67gfb bash
bash-4.4# cat /etc/mgmt/sharedsecret/key 
Bq0Qy9giIopNApzNaQvxzVIhW4Nh0rdI5PvkZksXzMJZnXknk18VAvB28u1QVEAHNJrervGKqY6Fq5OJMXF88VOT0YgPhgWL18EHzaEMy9mIwPRVatxzfHouSBYHFZ5xbash-4.4# 

10:~/EMCECS # k exec -it ecs-cluster-s3-7f4bc67458-tdmlv bash
Defaulting container name to s3.
Use 'kubectl describe pod/ecs-cluster-s3-7f4bc67458-tdmlv -n default' to see all of the containers in this pod.
bash-4.4# cd /etc/m
magic       mgmt/       modprobe.d/ motd        mtab        
bash-4.4# cd /etc/mgmt/sharedsecret/
bash-4.4# cat key 
Sg11zHt7ya1pGJ7ueYXp2Vru2YObRBQ4Uo8ouueBIfXIjAclVau6aYiATYF7nalgmvhvo1W8eGgGEytS4FQhfheNV0YtL7Eb9kvJUdCL7vYtR1tlWEgtjWnxETuLYwZybash-4.4# exit
exit
10:~/EMCECS # k exec -it ecs-cluster-geo-6bcf596465-wsmlv bash
Defaulting container name to geo.
Use 'kubectl describe pod/ecs-cluster-geo-6bcf596465-wsmlv -n default' to see all of the containers in this pod.
bash-4.4# cat /etc/mgmt/sharedsecret/key 
Sg11zHt7ya1pGJ7ueYXp2Vru2YObRBQ4Uo8ouueBIfXIjAclVau6aYiATYF7nalgmvhvo1W8eGgGEytS4FQhfheNV0YtL7Eb9kvJUdCL7vYtR1tlWEgtjWnxETuLYwZybash-4.4# 
bash-4.4# exit
exit
10:~/EMCECS # k exec -it ecs-cluster-geoservice-57ff9f846f-jhktg bash
Defaulting container name to geoservice.
Use 'kubectl describe pod/ecs-cluster-geoservice-57ff9f846f-jhktg -n default' to see all of the containers in this pod.
bash-4.4# cat /etc/mgmt/sharedsecret/key
Sg11zHt7ya1pGJ7ueYXp2Vru2YObRBQ4Uo8ouueBIfXIjAclVau6aYiATYF7nalgmvhvo1W8eGgGEytS4FQhfheNV0YtL7Eb9kvJUdCL7vYtR1tlWEgtjWnxETuLYwZybash-4.4# 
bash-4.4# exit
exit
10:~/EMCECS # k exec -it ecs-cluster-control-757ffbfd55-z5jsd bash
Defaulting container name to control.
Use 'kubectl describe pod/ecs-cluster-control-757ffbfd55-z5jsd -n default' to see all of the containers in this pod.
bash-4.4# cat /etc/mgmt/sharedsecret/key
Sg11zHt7ya1pGJ7ueYXp2Vru2YObRBQ4Uo8ouueBIfXIjAclVau6aYiATYF7nalgmvhvo1W8eGgGEytS4FQhfheNV0YtL7Eb9kvJUdCL7vYtR1tlWEgtjWnxETuLYwZybash-4.4#
```

File: ecs-cluster/templates/shared_secret.yaml (hunk 2 above)
Comment (eddiepavkovic) [reply ecs-cluster/templates/shared_secret.yaml:19]: ok looks good.

**Review by eddiepavkovic — APPROVED** [review]
ok

**Review by lipak2345 — APPROVED** [review]

**Pull request merged**